rasa/shared/core/domain.py

```diff
@@ -1954,6 +1956,36 @@ def _validate_forms(forms: Union[Dict, List], domain_slots: Dict[Text, Any]) ->
                     f"mapped in domain slots."
                 )
 
+        all_form_slots = [
+            required_slots[REQUIRED_SLOTS_KEY] for required_slots in forms.values()
+        ]
+        all_required_slots = set(chain.from_iterable(all_form_slots))
+
+        for slot, properties in domain_slots.items():
+            for mapping in properties["mappings"]:
+
+                if "conditions" in mapping:
+                    for condition in mapping["conditions"]:
+                        if (
+                            condition["active_loop"] == form_name
+                            and slot not in form_slots
+                        ):
+                            raise InvalidDomain(
+                                f"Slot '{slot}' has a mapping condition for form "
+                                f"'{form_name}', but it's not present in form "
+                                f"'{REQUIRED_SLOTS_KEY}'."
+                                f"The slot needs to be added to this key."
+                            )
+
+                if (
+                    mapping.get("type") == str(SlotMapping.FROM_TRIGGER_INTENT)
```

Comment:
btw, you can also call `.name` on the enums instead of doing the `str` cast

Reply:
Ah yes, but that actually returns the name in uppercase, while slot mappings in the domain are in lower case.